modules/wurflRtdProvider.js

```diff
@@ -0,0 +1,234 @@
+import { submodule } from '../src/hook.js';
+import { fetch } from '../src/ajax.js';
+import {
+  mergeDeep,
+  prefixLog,
+} from '../src/utils.js';
+
+// Constants
+const REAL_TIME_MODULE = 'realTimeData';
+const MODULE_NAME = 'wurfl';
+
+// WURFL_JS_HOST is the host for the WURFL service endpoints
+const WURFL_JS_HOST = 'https://prebid.wurflcloud.com';
+// WURFL_JS_ENDPOINT_PATH is the path for the WURFL.js endpoint used to load WURFL data
+const WURFL_JS_ENDPOINT_PATH = '/wurfl.js';
+// STATS_ENDPOINT_PATH is the path for the stats endpoint used to send analytics data
+const STATS_ENDPOINT_PATH = '/v1/prebid/stats';
+// WURFL_JS_TIMEOUT is the default timeout for WURFL.js to load in milliseconds
+const WURFL_JS_TIMEOUT = 500;
+
+const logger = prefixLog('[WURFL RTD Submodule]');
+
+// enrichedBidders holds a list of prebid bidder names, of bidders which have been
+// injected with WURFL data
+const enrichedBidders = new Set();
+
+/**
+ * init initializes the WURFL RTD submodule
+ * @param {Object} config Configuration for WURFL RTD submodule
+ * @param {Object} userConsent User consent data
+ */
+const init = (config, userConsent) => {
+  logger.logMessage('initialized');
+  return true;
+}
+
+/**
+ * getBidRequestData enriches the OpenRTB 2.0 device data with WURFL data
+ * @param {Object} reqBidsConfigObj Bid request configuration object
+ * @param {Function} callback Called on completion
+ * @param {Object} config Configuration for WURFL RTD submodule
+ * @param {Object} userConsent User consent data
+ */
+const getBidRequestData = (reqBidsConfigObj, callback, config, userConsent) => {
+  const altHost = config.params?.altHost ?? null;
+  const isDebug = config.params?.debug ?? false;
+
+  const bidders = new Set();
+  reqBidsConfigObj.adUnits.forEach(adUnit => {
+    adUnit.bids.forEach(bid => {
+      bidders.add(bid.bidder);
+    });
+  });
+
+  let host = WURFL_JS_HOST;
+  if (altHost) {
+    host = altHost;
+  }
+
+  const wjsPromise = new Promise((resolve, reject) => {
+    const url = new URL(host);
+    url.pathname = WURFL_JS_ENDPOINT_PATH;
+
+    const runWjs = (src) => {
+      const wjs = document.createElement('script');
```

Comment:
You add yourself to adloader but then you didn't import and use that method

Reply:
Updated in 32d1c0e

Reply:
Doesn't appear fixed, you have to user the loadexternalscript method

Reply:
Updated to use loadExternalScript in 1bb2df6